tests/std/tests/P0896R4_views_split/test.cpp

```diff
@@ -328,6 +329,32 @@ constexpr bool test_devcom_1559808() {
     return true;
 }
 
+struct CopyConstructibleRange {
+    constexpr CopyConstructibleRange(vector<int> v) : inner(move(v)) {}
+    CopyConstructibleRange(const CopyConstructibleRange&) = default;
+    CopyConstructibleRange& operator=(const CopyConstructibleRange&) = delete;
+    constexpr const int* begin() const {
+        return inner.data();
+    }
+    constexpr const int* end() const {
+        return inner.data() + inner.size();
+    }
+
+    vector<int> inner;
+};
+
+constexpr bool test_LWG_3590() {
+    CopyConstructibleRange r{{1, 2, 3, 4}};
+    const auto split_view = views::split(r, 0);

```

Comment:
No change requested: I observe that `split_view` and `ref_view` below are named identically to their types - however, they aren't technically shadowing as this test isn't `using namespace std::ranges;` so it's fine.